src/reducers/Material.js

```diff
@@ -43,8 +43,14 @@ function materialsToggleIsNonPeriodicForOne(state, action) {
         return state;
     }
     newMaterial.isNonPeriodic = !newMaterial.isNonPeriodic;
-    Made.tools.material.scaleLatticeToMakeNonPeriodic(newMaterial);
-    Made.tools.material.getBasisConfigTranslatedToCenter(newMaterial);
+    if (newMaterial.isNonPeriodic) {
+        newMaterial.toCartesian()
+        Made.tools.material.scaleLatticeToMakeNonPeriodic(newMaterial);
+        Made.tools.material.getBasisConfigTranslatedToCenter(newMaterial);
+    } else {
+        newMaterial.toCrystal()
+    }
```

Comment:
I don't understand why we need the above logic. `Made.tools.material.scaleLatticeToMakeNonPeriodic` should do nothing with the structure when passed a structure that is already non-periodic. Same for getBasisConfigTranslatedToCenter - repeated application of the function should not be doing anything

Reply:
Please see my comment on the Jira board. https://exabyte.atlassian.net/browse/SOF-5478#:~:text=I%20think%20the,non%2Dperiodic%20structures.